scene/gui/item_list.cpp

```diff
@@ -295,35 +295,19 @@ int ItemList::get_current() const {
 	return current;
 }
 
-void ItemList::move_item(int p_item, int p_to_pos) {
+void ItemList::move_item(int p_from_idx, int p_to_idx) {
 
-	ERR_FAIL_INDEX(p_item, items.size());
-	ERR_FAIL_INDEX(p_to_pos, items.size() + 1);
+	ERR_FAIL_INDEX(p_from_idx, items.size());
+	ERR_FAIL_INDEX(p_to_idx, items.size());
 
-	Item it = items[p_item];
-	items.remove(p_item);
+	Item item = items[p_from_idx];
+	items.remove(p_from_idx);
+	items.insert(p_to_idx, item);
 
-	if (p_to_pos > p_item) {
-		p_to_pos--;
-	}
-
-	if (p_to_pos >= items.size()) {
-		items.push_back(it);
-	} else {
-		items.insert(p_to_pos, it);
-	}
-
-	if (current < 0) {
-		//do none
-	} else if (p_item == current) {
-		current = p_to_pos;
-	} else if (p_to_pos > p_item && current > p_item && current < p_to_pos) {
-		current--;
-	} else if (p_to_pos < p_item && current < p_item && current > p_to_pos) {
-		current++;
-	}
+	current = p_to_idx;
```

Comment:
The previous code tried to keep the current selection active, so that if you move an item which is not the select one, you will keep the existing selection instead of picking the moved one.

Since that code was never exposed we can change it, but what would be the most expected behaviour?

Reply:
It is a feature that doesn't interfere. I think this should work now too.

Reply:
`current` is an int. If `current` is 3, you remove item 1, and then insert it back at position 5, you will go from:
```
0 item0
1 item1
2 item2
*3 item3
4 item4
5 item5
6 item6
```
to
```
0 item0
1 item2
2 item3
*3 item4  // selection kept at "3", but new item selected
4 item5
5 item6
6 item1  // moved
```
(or maybe 5 & 6 should be swapped in my example, not sure if `insert` goes before or after the index, but it doesn't matter here).